packages/firestore/src/platform_browser/browser_platform.ts

```diff
@@ -72,4 +73,27 @@ export class BrowserPlatform implements Platform {
   btoa(raw: string): string {
     return btoa(raw);
   }
+
+  randomBytes(nBytes: number): Uint8Array {
+    if (nBytes <= 0) {
+      return new Uint8Array();
+    }
+
+    // Polyfill for IE and WebWorker
+    // eslint-disable-next-line no-restricted-globals
+    const crypto =
+      (this.window &&
+        (this.window.crypto || (this.window as any)['msCrypto'])) ||
+      (self && self.crypto);
```

Comment:
According to https://developer.mozilla.org/en-US/docs/Web/API/Window/self, you can just write:

```
const crypto = typeof self !=== 'undefined' && (self.crypto || (self as any)['msCrypto']);
```

Reply:
Done.